src/utils/data_reader.py

```
@@ -0,0 +1,136 @@
+        if source not in ['openmeteo', 'nasapower']:
+            raise ValueError(f"Invalid weather source: {source}")
+        
+        files = self.manager.get_date_range_files(
+            source, start_date, end_date, country
+        )
+        
+        if not files:
+            raise FileNotFoundError(
+                f"No {source} data found for {country} "
+                f"between {start_date} and {end_date}"
+            )
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)